File: core/src/com/jediterm/core/compatibility/Point.java
package com.jediterm.core.compatibility;

import java.util.Objects;

public final class Point {
  public int x;
  public int y;

  public Point() {
    this(0, 0);
  }

  public Point(int x, int y) {
    setLocation(x, y);
  }

  public Point(Point other) {
    setLocation(other);
  }

  public void setLocation(int x, int y) {
    this.x = x;
    this.y = y;
  }

  public void setLocation(Point p) {
    x = p.x;
    y = p.y;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof Point)) return false;
    Point point = (Point) o;
    return x == point.x && y == point.y;
  }

  @Override
  public int hashCode() {
    return Objects.hash(x, y);
  }

  public String toString() {
    return "[x=" + x + ",y=" + y + "]";
  }
}

File: jediterm-core-mpp/src/jvmMain/kotlin/com/jediterm/core/compatibility/Point.kt
package com.jediterm.core.compatibility

import java.util.*

class Point {
    @JvmField
    var x: Int = 0
    @JvmField
    var y: Int = 0

    @JvmOverloads
    constructor(x: Int = 0, y: Int = 0) {
        setLocation(x, y)
    }

    constructor(other: Point) {
        setLocation(other)
    }

    fun setLocation(x: Int, y: Int) {
        this.x = x
        this.y = y
    }

    fun setLocation(p: Point) {
        x = p.x
        y = p.y
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o !is Point) return false
        val point = o
        return x == point.x && y == point.y
    }

    override fun hashCode(): Int {
        return Objects.hash(x, y)
    }

    override fun toString(): String {
        return "[x=$x,y=$y]"
    }
}
